Refactor stuff

PieterVdc opened the pull request:
split off from the #2043 Lua PR, mostly reordering some things to make it easier to use with lua later on

PR shouldn't have any functional changes

Loobinex commented:
@PieterVdc I was gonna test and merge this in a hurry, but it has merge conflicts.

PieterVdc commented:
> > @PieterVdc I was gonna test and merge this in a hurry, but it has merge conflicts.
> 
> I fixed those conflicts since I introduced them with #3820.

perfect thanks, also why I did this PR in the first place, to have less new ones in the future on my lua PR

PieterVdc commented on src/magic.c:
```diff
@@ -2195,6 +2196,131 @@ TbBool update_power_overcharge(struct PlayerInfo *player, int pwkind)
+ * @param fmcl_bytes encoded bytes: f=cast for free flag,m=spell kind,c=caster player index,l=spell level.
+ * @return TbResult whether the spell was successfully cast
+ */
+TbResult script_use_spell_on_creature(PlayerNumber plyr_idx, ThingModel crmodel, long criteria, long fmcl_bytes)
```
hmm yes makes sense, I'll move it

PieterVdc replied:
I suppose renaming magic.c to keeper_powers.c or something might also make sense

PieterVdc commented:
> @PieterVdc I was gonna test and merge this in a hurry, but it has merge conflicts.

all good now
